assets/js/interactivity/store.js

```diff
@@ -32,12 +32,20 @@ const getSerializedState = () => {
 	return {};
 };
 
+const storeCallbacks = {};
 const rawState = getSerializedState();
 export const rawStore = { state: deepSignal( rawState ) };
 
 if ( typeof window !== 'undefined' ) window.store = rawStore;
 
-export const store = ( { state, ...block } ) => {
+export const store = ( { state, ...block }, callbacks = {} ) => {
 	deepMerge( rawStore, block );
 	deepMerge( rawState, state );
+	Object.entries( callbacks ).forEach( ( [ key, cb ] ) => {
+		( storeCallbacks[ key ] = storeCallbacks[ key ] || [] ).push( cb );
+	} );
+};
+
+export const runStoreCallbacks = ( key ) => {
+	storeCallbacks[ key ]?.forEach( ( cb ) => cb( rawStore ) );
```

Comment:
I don't expect to have an unlimited number of callbacks so I think we can hardcode their names. I also don't intend to use the second parameter only for callbacks, but for options in general.

Also, I'm not 100% sure about it yet, but maybe instead of an array, we could use a `Set` to avoid adding the same callback more than once?

Something like this:

```js
const callbacks = {
  afterLoad: new Set(),
};

export const store = ({ state, ...block } = {}, options = {}) => {
  deepMerge(rawStore, block);
  deepMerge(rawState, state);
  if (options.afterLoad) callbacks.afterLoad.add(options.afterLoad);
};

export const runStoreCallbacks = (key) => {
  callbacks[key]?.forEach((cb) => cb(rawStore));
};
```